### PR TITLE
fix: migrated sidelist to tsx

### PR DESCRIPTION
**Pull request by aizad-deriv**
## Changes:

Please include a summary of the change and which issue is fixed below:
-   ...

## When you need to add unit test

-   [ ] If this change disrupt current flow
-   [ ] If this change is adding new flow

## When you need to add integration test

-   [ ] If components from external libraries are being used to define the flow, e.g. @deriv/components
-   [ ] If it relies on a very specific set of props with no default behavior for the current component.

## Test coverage checklist (for reviewer)

-   [ ] Ensure utility / function has a test case 
-   [ ] Ensure all the tests are passing

## Type of change

-   [ ] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [ ] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

File: packages/reports/src/Components/Form/CompositeCalendar/side-list.tsx [hunk 1]
```diff
@@ -1,16 +1,28 @@
...
-const isActive = (from, to, flag) => {
+type TItem = {
+    value: string;
+    label: string;
+    onClick: () => void;
+    duration: number;
+};
+
+type TSideList = {
+    from: string;
```
Comment (jim-deriv) [inline packages/reports/src/Components/Form/CompositeCalendar/side-list.tsx]: @aizad-deriv Why did you change this from the original data type (`number`) to `string`?
```suggestion
    from: number;
```

File: packages/reports/src/Components/Form/CompositeCalendar/side-list.tsx [hunk 2]
```diff
@@ -1,16 +1,28 @@
...
+    items: Array<TItem>;
+    to: number;
+};
+
+const isActive = (from: string, to: number, flag: number) => {
```
Comment (jim-deriv) [inline packages/reports/src/Components/Form/CompositeCalendar/side-list.tsx]:
```suggestion
const isActive = (from: number, to: number, flag: number) => {
```

File: packages/reports/src/Components/Form/CompositeCalendar/side-list.tsx (hunk 1 above)
Comment (aizad-deriv) [reply packages/reports/src/Components/Form/CompositeCalendar/side-list.tsx]: I will change it 🙇🏼  back to the original

File: packages/reports/src/Components/Form/CompositeCalendar/side-list.tsx (hunk 2 above)
Comment (aizad-deriv) [reply packages/reports/src/Components/Form/CompositeCalendar/side-list.tsx]: I will change it 🙇🏼  back to the original

File: packages/reports/src/Components/Form/CompositeCalendar/side-list.tsx (hunk 1 above)
Comment (jim-deriv) [reply packages/reports/src/Components/Form/CompositeCalendar/side-list.tsx]: prop-type

File: packages/reports/src/Components/Form/CompositeCalendar/side-list.tsx [hunk 3]
```diff
@@ -1,16 +1,28 @@
...
 import React from 'react';
 import { toMoment } from '@deriv/shared';
 import ListItem from './list-item.jsx';
 
-const isActive = (from, to, flag) => {
+type TItem = {
+    value: string;
+    label: string;
+    onClick: () => void;
+    duration: number;
```
Comment (jim-deriv) [inline packages/reports/src/Components/Form/CompositeCalendar/side-list.tsx:9]: `
type TItem = {
    value: string;
    label: string;
    onClick: () => void;
    duration: number;
`

@aizad-deriv, somehow, I think this type needs to sync with the one for `ListItem`, you might get errors when the 2 PRs are merged together.

**Review by jim-deriv — APPROVED** [review]

**Review by niloofar-deriv — APPROVED** [review]

**Review by akmal-deriv — APPROVED** [review]
